src/transformStateWithClones.js

```diff
@@ -7,7 +7,46 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const options = Object.values(actions);
+  const result = [];
+
+  for (const option of options) {
+    const { keysToRemove, type, extraData } = option;
+    const stateCopy = { ...state };
+
+    switch (type) {
+      case 'addProperties':
+        let lastElementOfArray = null;
+
+        if (result[result.length - 1]) {
+          lastElementOfArray = Object.assign({}, result[result.length - 1],
+            extraData);
+        } else {
+          lastElementOfArray = Object.assign(stateCopy, extraData);
+        }
+        result.push(lastElementOfArray);
+        break;
+
+      case 'removeProperties':
+        let lastElementToRemove = null;
```

Comment:
same here. you hardcoded it a lot, please simplify this logic as it is really hard to understand